### DISCUSSION TITLE
[help] Opposite of tar_invalidate

### DISCUSSION THREAD

**Discussion by mem48** [ropensci/targets#1274] [Help]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

Is there a way to do the opposite of `tar_invalidate()` i.e. make a target that is outdated up to date without rerunning it via `tar_make()`? I can't find any documentation on this issue. 

My use case is that I have a helper function e.g. `read_data()` that is used by many targets.
```
read_data = function(path){
    read.csv(path)
}
```
At some point, I update this function with some new features. 
```
read_data = function(path, sep = ","){
    read.csv(path, sep  = sep )
}
```
Now all my targets that use this function are outdated. But I know that this new feature is not relevant to the old targets. In this case, the new `sep` argument is the default for `read.csv` so the result of `read_data("my.csv")` will be identical with the old and new functions.

Now I could just run `tar_make()` but suppose that I have a large workflow with hundreds of files that are GB in size, and have plenty of downstream processing. This could be hours or days of computation to recreate the targets that I already have.

There should be a simple way to say "this outdated target is actually up to date".

**Comment by joelnitta** [comment]
If the output of the new function is indeed exactly the same, you are correct that the `read_data()` function will run again, but **no other downstream dependencies will trigger**.

It may seem this way when you look at the dependency graph, but that is because {targets} has no way to know what the output will be until it actually runs the code. The "outdated" targets in the dependency graph are only estimates. The graph is accurate in terms of the dependency relationships, but it cannot actually predict precisely which targets will be run with the next `tar_make()`.

**Reply by wlandau** [reply]
Yes, since `targets` uses hashes, it should not does not rebuild anything downstream of the targets that call `read_data()`. If you want to skip those upstream targets too, you could consider `cue = tar_cue(depend = FALSE)` in `tar_target()`, but this is a bit of a blunt instrument and best as a temporary workaround only.
